cpg_infra/billing_aggregator/monthly_aggregate/main.py

```diff
@@ -32,10 +34,63 @@ def get_bigquery_client() -> bq.Client:
     return bq.Client()
 
 
-def from_request(*args, **kwargs):
+def get_invoice_month_from_request(
+    request: Request,
+) -> str | None:
+    """
+    Get the invoice month from the cloud function request.
+    """
+    if not request:
+        print('No request found')
+        return None
+
+    content_type = request.content_type
+    if request.method == 'GET':
+        print('GET request, using args')
+        request_data = request.args
+    elif content_type == 'application/json':
+        print('JSON found in request')
+        request_data = request.get_json(silent=True)
+    elif content_type in ('application/octet-stream', 'text/plain'):
+        print('Text data found')
+        request_data = json.loads(request.data)
+    elif content_type == 'application/x-www-form-urlencoded':
+        print('Encoded Form')
+        request_data = request.form
+    else:
+        raise ValueError(f'Unknown content type: {content_type}')
+
+    if 'attributes' in request_data and 'invoice_month' in request_data.get(
+        'attributes'
+    ):
+        request_data = request_data['attributes']
+    elif 'message' in request_data and 'data' in request_data.get('message'):
+        try:
+            request_data = json.loads(b64decode(request_data['message']['data']))
+        except Exception as exp:
+            raise exp
+
+    print(request_data)
+
+    if request_data and 'invoice_month' in request_data:
+        invoice_month = request_data.get('invoice_month')
+    else:
+        raise ValueError("JSON is invalid, or missing a 'invoice_month'")
+
+    return invoice_month
+
+
+@functions_framework.http
+def from_request(request: Request):
     """Entrypoint for cloud functions, run always as default (previous month)"""
-    asyncio.new_event_loop().run_until_complete(
-        process_and_upload_monthly_billing_report(None)
+
+    try:
+        invoice_month = get_invoice_month_from_request(request)
+    except ValueError:
+        invoice_month = None
```

Comment:
Would be worth logging the ValueError here and indicating that you're setting it to None (the default)